core/UITreeGenerator.py

```diff
@@ -65,25 +66,32 @@ def generate_ui_subtree(self, fhir_profile_snapshot: dict, module_name) -> List[
     def translate(self, fhir_profile_snapshot: dict, applicable_querying_meta_data: List[ResourceQueryingMetaData]) \
             -> List[TreeMap]:
         """
-        Translates the given FHIR profile snapshot into a ui tree
+        Translates the given FHIR profile snapshot into a UI tree
         :param fhir_profile_snapshot: FHIR profile snapshot json representation
-        :param applicable_querying_meta_data: applicable querying meta data
+        :param applicable_querying_meta_data: applicable querying metadata
         :return: root of the ui tree
         """
-        result: List[TreeMap] = []
+        result_map: dict[(TermCode, str), TreeMap] = dict()
         for applicable_querying_meta_data in applicable_querying_meta_data:
             # TODO: add context information to the ui tree
```

Comment:
@paulolaup  as the context has been added this todo should probably be removed - double check and if not needed anymore - remove